tensorflow/python/util/tf_inspect.py

```diff
@@ -45,6 +45,27 @@ def getargspec(object):  # pylint: disable=redefined-builtin
                if d.decorator_argspec is not None), _inspect.getargspec(target))
 
 
+def getfullargspec(object):  # pylint: disable=redefined-builtin
+  """TFDecorator-aware replacement for inspect.getargspec.
+
+  Args:
+    object: A callable, possibly decorated.
+
+  Returns:
+    The `FullArgSpec` (Python 3) or `ArgSpec` (Python 2)
```

Comment:
@martinwicke They are not truly compatible as FullArgSpec has `varkw` while ArgSpec has `keywords`. For example, if `getfullargspec` is used, then in `tensorflow/python/util/deprecation.py`:

https://github.com/tensorflow/tensorflow/blob/325fc2330f4e527b394456ce1d821f2c73b1e7a2/tensorflow/python/util/deprecation.py#L224-L229

The following has to be used:
```
# Cannot use arg_spec.keywords directory
getattr(arg_spec, "keywords", getattr(arg_spec, "varkw"))
```

For that I try to use `util.getfullargspec` only for `estimator.Estimator`. For everything else, the original `util.getargspec` would be used.